File: interactive.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

/**
 * Ask and remember a user's name and say hello
 *
 * Run as (with gcc):
 *
 * c++ -o interactive interactive.cpp
 * ./interactive
 *
 * Or with clang:
 *
 * clang -x c++ -o interactive interactive.cpp
 * ./interactive
 */
int main()
{
    string value;
    vector<string> names;
    bool running = true;

    while (running)
    {
        cout << "Your name: ";
        cin >> value;

        if (value == "exit") {
            running = false;
            cout << "Bye!\n";
        } else if (find(names.begin(), names.end(), value) != names.end()) {
            cout << "Hello " << value << ", we have met before :)\n";
        } else {
            names.push_back(value);
            cout << "Hi " << value << "!\n";
        }
    }
}

File: interactive.py
"""Ask and remember a user's name and say hello

Note: this depends on the name of your python executable, this example assumes GNU/Linux python3

Run as:

python3 interactive.py
"""

names = []
running = True

while running:
    value = input("Your name: ")

    if value == "exit":
        running = False
        print("Bye!")
    elif value in names:
        print(f"Hello {value}, we have met before :)")
    else:
        names.append(value)
        print(f"Hi {value}!")
